lib/jekyll/frontmatter_defaults.rb

```diff
@@ -100,12 +100,27 @@ def applies?(scope, path, type)
     def applies_path?(scope, path)
       return true if !scope.key?("path") || scope["path"].empty?
 
-      scope_path = Pathname.new(scope["path"])
-      Pathname.new(sanitize_path(path)).ascend do |ascended_path|
-        if ascended_path.to_s == scope_path.to_s
+      sanitized_path = sanitize_path(path)
+
+      site_path = Pathname.new(@site.source)
+      rel_scope_path = Pathname.new(scope["path"])
+      abs_scope_path = File.join(@site.source, rel_scope_path)
+      Dir.glob(abs_scope_path).each do |scope_path|
```

Comment:
Looked through that code, it relies on `File#fnmatch?` that uses the same rules as `Dir#glob`, though `EntryFilter#glob_include?` itself is not super-useful in this code, it would take more lines of code to reuse `glob_include?`